src/core/main.c

```diff
@@ -1884,7 +1884,8 @@ static int do_reexecute(
         if (switch_root_dir) {
                 r = switch_root(/* new_root= */ switch_root_dir,
                                 /* old_root_after= */ NULL,
-                                /* flags= */ objective == MANAGER_SWITCH_ROOT ? SWITCH_ROOT_DESTROY_OLD_ROOT : 0);
+                                /* flags= */ (objective == MANAGER_SWITCH_ROOT ? SWITCH_ROOT_DESTROY_OLD_ROOT : 0) |
+                                             (objective == MANAGER_SOFT_REBOOT ? SWITCH_ROOT_NON_RECURSIVE_RUN : 0));
```

Comment:
We usually name thing in positive, So, I'd like the following.
```
(objective == MANAGER_SWITCH_ROOT ? SWITCH_ROOT_DESTROY_OLD_ROOT | SWITCH_ROOT_RECURSIVE_RUN : 0)
```

Reply:
that requires changing more places, so I'll leave it as-is

Reply:
> that requires changing more places, so I'll leave it as-is

More places? The enum is introduced by the commit. I do not think that's so many. Only three places in your patch.

Reply:
switch_root() is used elsewhere too

Reply:
I checked now, and it's used in exactly two places ;)  So I think it'd be nicer to rework the flag to be positive. The definition of `run_mount_flags` below becomes much nicer with a positive option.